src/videojs-contrib-hls.js

```diff
@@ -134,27 +134,73 @@ export default class HlsHandler extends Component {
       this.tech_.audioTracks().addEventListener('change', this.audioTrackChange_);
     });
 
-    this.masterPlaylistController_.on('selectedinitialmedia', () => {
+    this.masterPlaylistController_.on('audioinfochanged', (e) => {
+      if (!videojs.browser.IS_FIREFOX) {
+        return;
+      }
       let audioTrackList = this.tech_.audioTracks();
-      let media = this.masterPlaylistController_.media();
-      let master = this.masterPlaylistController_.master();
-      let mediaGroups = master.mediaGroups;
-      let attributes = {
-        audio: {main: {default: true}}
-      };
-
-      if (!media.attributes) {
-        // source URL was playlist manifest, not master
-        // no audio tracks to add
+      let error = 'had different audio properties (channels, sample rate, etc.) ' +
+                  'or changed in some other way.  This behavior is currently ' +
+                  'unsupported in Firefox due to an issue: \n\n' +
+                  'https://bugzilla.mozilla.org/show_bug.cgi?id=1247138\n\n';
+
+      let attributes = this.masterPlaylistController_.getPlaylistAttributes_();
+      let mainLabel;
+
+      for (let label in attributes.audio) {
+        let hlstrack = attributes.audio[label];
+
+        /* eslint-disable dot-notation */
+        // dot notation used on the key default to support ie8
+        if (hlstrack['default']) {
+          mainLabel = label;
+          break;
+        }
+        /* eslint-enable dot-notation */
+      }
+
+      let trackToRemove;
+      let mainTrack;
+
+      for (let i = 0; i < audioTrackList.length; i++) {
+        let track = audioTrackList[i];
+
+        if (track.enabled) {
+          trackToRemove = track;
+        } else if (track.label === mainLabel) {
```

Comment:
I guess it would signal a bigger problem if the main track caused the change and had the issue. If it is possible to have that condition though, it looks like this might throw an error (down when we call mainTrack.enabled).